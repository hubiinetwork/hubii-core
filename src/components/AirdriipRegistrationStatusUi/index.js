/**
*
* AirdriipRegistrationStatusUi
*
*/

import React from 'react';
import { Spin, Alert, Icon } from 'antd';
import PropTypes from 'prop-types';
import Text from 'components/ui/Text';

const AirdriipRegistrationStatusUi = ({ status, loading, style }) => {
  if (loading) {
    return (
      <div
        style={{
          ...style,
          display: 'flex',
          justifyContent: 'center',
        }}
      >
        <Text>{'Checking registration status'}</Text>
        <Spin
          style={{ marginLeft: '0.75rem' }}
          indicator={<Icon type="loading" />}
        />
      </div>
    );
  }
  if (status === 'registered') {
    return (
      <Alert
        message="Address registered"
        description="No further action is required"
        type="success"
        showIcon
        style={style}
      />
    );
  }
  if (status === 'unregistered') {
    return (
      <Alert
        message="Address is not yet registered"
        type="info"
        showIcon
        style={style}
      />
    );
  }
  return (
    <Alert
      message="There was a problem communicating with the nahmii airdriip service, please try again later."
      description={status}
      type="error"
      showIcon
      style={style}
    />
  );
};

AirdriipRegistrationStatusUi.propTypes = {
  loading: PropTypes.bool.isRequired,
  status: PropTypes.string,
  style: PropTypes.object,
};

export default AirdriipRegistrationStatusUi;
