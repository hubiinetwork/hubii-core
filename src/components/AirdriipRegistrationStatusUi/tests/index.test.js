import React from 'react';
import { shallow } from 'enzyme';

import AirdriipRegistrationStatusUi from '../index';

describe('<AirdriipRegistrationStatusUi />', () => {
  let props;
  beforeEach(() => {
    props = {
      loading: true,
      style: { marginBottom: '5px' },
    };
  });
  it('should render correctly in loading state', () => {
    const wrapper = shallow(<AirdriipRegistrationStatusUi {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly in registered state', () => {
    const wrapper = shallow(
      <AirdriipRegistrationStatusUi
        {...props}
        status="registered"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly in error state', () => {
    const wrapper = shallow(
      <AirdriipRegistrationStatusUi
        {...props}
        status="Error: asdkjfasjdkl"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
