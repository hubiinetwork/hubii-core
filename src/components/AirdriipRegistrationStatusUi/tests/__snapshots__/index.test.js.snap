// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AirdriipRegistrationStatusUi /> should render correctly in error state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AirdriipRegistrationStatusUi
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    loading={true}
    status="Error: asdkjfasjdkl"
    style={
        Object {
            "marginBottom": "5px",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Text>
          checking_registration_status
</Text>,
        <Spin
          indicator={
                    <Icon
                              type="loading"
                    />
          }
          prefixCls="ant-spin"
          size="default"
          spinning={true}
          style={
                    Object {
                              "marginLeft": "0.75rem",
                            }
          }
          wrapperClassName=""
/>,
      ],
      "style": Object {
        "display": "flex",
        "justifyContent": "center",
        "marginBottom": "5px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "checking_registration_status",
        },
        "ref": null,
        "rendered": "checking_registration_status",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "indicator": <Icon
            type="loading"
/>,
          "prefixCls": "ant-spin",
          "size": "default",
          "spinning": true,
          "style": Object {
            "marginLeft": "0.75rem",
          },
          "wrapperClassName": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Text>
            checking_registration_status
</Text>,
          <Spin
            indicator={
                        <Icon
                                    type="loading"
                        />
            }
            prefixCls="ant-spin"
            size="default"
            spinning={true}
            style={
                        Object {
                                    "marginLeft": "0.75rem",
                                  }
            }
            wrapperClassName=""
/>,
        ],
        "style": Object {
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "5px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "checking_registration_status",
          },
          "ref": null,
          "rendered": "checking_registration_status",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "indicator": <Icon
              type="loading"
/>,
            "prefixCls": "ant-spin",
            "size": "default",
            "spinning": true,
            "style": Object {
              "marginLeft": "0.75rem",
            },
            "wrapperClassName": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<AirdriipRegistrationStatusUi /> should render correctly in loading state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AirdriipRegistrationStatusUi
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    loading={true}
    style={
        Object {
            "marginBottom": "5px",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Text>
          checking_registration_status
</Text>,
        <Spin
          indicator={
                    <Icon
                              type="loading"
                    />
          }
          prefixCls="ant-spin"
          size="default"
          spinning={true}
          style={
                    Object {
                              "marginLeft": "0.75rem",
                            }
          }
          wrapperClassName=""
/>,
      ],
      "style": Object {
        "display": "flex",
        "justifyContent": "center",
        "marginBottom": "5px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "checking_registration_status",
        },
        "ref": null,
        "rendered": "checking_registration_status",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "indicator": <Icon
            type="loading"
/>,
          "prefixCls": "ant-spin",
          "size": "default",
          "spinning": true,
          "style": Object {
            "marginLeft": "0.75rem",
          },
          "wrapperClassName": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Text>
            checking_registration_status
</Text>,
          <Spin
            indicator={
                        <Icon
                                    type="loading"
                        />
            }
            prefixCls="ant-spin"
            size="default"
            spinning={true}
            style={
                        Object {
                                    "marginLeft": "0.75rem",
                                  }
            }
            wrapperClassName=""
/>,
        ],
        "style": Object {
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "5px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "checking_registration_status",
          },
          "ref": null,
          "rendered": "checking_registration_status",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "indicator": <Icon
              type="loading"
/>,
            "prefixCls": "ant-spin",
            "size": "default",
            "spinning": true,
            "style": Object {
              "marginLeft": "0.75rem",
            },
            "wrapperClassName": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<AirdriipRegistrationStatusUi /> should render correctly in registered state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AirdriipRegistrationStatusUi
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    loading={true}
    status="registered"
    style={
        Object {
            "marginBottom": "5px",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Text>
          checking_registration_status
</Text>,
        <Spin
          indicator={
                    <Icon
                              type="loading"
                    />
          }
          prefixCls="ant-spin"
          size="default"
          spinning={true}
          style={
                    Object {
                              "marginLeft": "0.75rem",
                            }
          }
          wrapperClassName=""
/>,
      ],
      "style": Object {
        "display": "flex",
        "justifyContent": "center",
        "marginBottom": "5px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "checking_registration_status",
        },
        "ref": null,
        "rendered": "checking_registration_status",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "indicator": <Icon
            type="loading"
/>,
          "prefixCls": "ant-spin",
          "size": "default",
          "spinning": true,
          "style": Object {
            "marginLeft": "0.75rem",
          },
          "wrapperClassName": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Text>
            checking_registration_status
</Text>,
          <Spin
            indicator={
                        <Icon
                                    type="loading"
                        />
            }
            prefixCls="ant-spin"
            size="default"
            spinning={true}
            style={
                        Object {
                                    "marginLeft": "0.75rem",
                                  }
            }
            wrapperClassName=""
/>,
        ],
        "style": Object {
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "5px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "checking_registration_status",
          },
          "ref": null,
          "rendered": "checking_registration_status",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "indicator": <Icon
              type="loading"
/>,
            "prefixCls": "ant-spin",
            "size": "default",
            "spinning": true,
            "style": Object {
              "marginLeft": "0.75rem",
            },
            "wrapperClassName": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
